src/stratis/run.rs

```diff
@@ -72,16 +74,29 @@ pub fn run(sim: bool) -> StratisResult<()> {
         };
 
         let (trigger, should_exit) = channel(1);
-        let (sender, receiver) = unbounded_channel::<UdevEngineEvent>();
+        let (udev_sender, udev_receiver) = unbounded_channel::<UdevEngineEvent>();
+        #[cfg(feature = "dbus_enabled")]
+        let (dbus_sender, dbus_receiver) = unbounded_channel::<DbusAction>();
 
-        let join_udev = task::spawn_blocking(move || udev_thread(sender, should_exit));
-        let join_ipc = task::spawn(setup(engine.clone(), receiver, trigger.clone()));
+        let join_udev = task::spawn_blocking(move || udev_thread(udev_sender, should_exit));
+        let join_ipc = task::spawn(setup(
+            engine.clone(),
+            udev_receiver,
+            #[cfg(feature = "dbus_enabled")]
```

Comment:
The only reason trigger exists is for blocking threads in the D-Bus API. Our JSON RPC implementation does not block on futures so they will be cancelled once they yield, rendering the trigger useless. I was trying to keep the method signature identical to avoid conditional compilation, but now that they've diverged in the PR, it seems reasonable to not pass this in any longer.